### PR TITLE
[NPU] npu support `transpose`

### PR DESCRIPTION
**Pull request by Meiyim**
### PR types
New features
### PR changes
OPs

### Describe

+ support npu kernel for `transpose`

python unitest:

901dc4a9aecdf46c2f0981594

infoflow 2021-03-16 16-18-11

### CODE REVIEW

File: paddle/fluid/operators/transpose_op_npu.cc [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    void Compute(const framework::ExecutionContext& ctx) const override {
+        auto* x = ctx.Input<framework::LoDTensor>("X");
+        auto* out = ctx.Output<framework::LoDTensor>("Out");
+        std::vector<int> axis = ctx.Attr<std::vector<int>>("axis");
+        framework::NPUAttributeMap attr_input = {{"perm", axis}};
+        out->mutable_data<T>(ctx.device_context().GetPlace());
+        auto runner = NpuOpRunner("TransposeD", {*x}, {*out}, attr_input);
+        auto stream = ctx.template device_context<paddle::platform::NPUDeviceContext>().stream();
+        runner.Run(stream);
+        //.Wait();
```
Comment (zhiqiu) [inline paddle/fluid/operators/transpose_op_npu.cc]: Remove usunsed code.

Comment (Meiyim) [reply paddle/fluid/operators/transpose_op_npu.cc]: removed

File: python/paddle/fluid/tests/unittests/npu/test_transpose_op_npu.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+    def init_kernel_type(self):
+        self.use_mkldnn = False
+
+    def init_input_output(self):
+        self.x = np.random.uniform(0.1, 1, [8, 512, 12, 64]).astype(self.dtype)
+        self.out = np.transpose(self.x, [0, 2, 1, 3])
+
+    def init_dtype(self):
+        self.dtype = np.float32
```
Comment (liym27) [inline python/paddle/fluid/tests/unittests/npu/test_transpose_op_npu.py:55]: Add a unittest for `float16`

**Review by liym27 — CHANGES_REQUESTED** [review]

File: python/paddle/fluid/tests/unittests/npu/test_transpose_op_npu.py (hunk 2 above)
Comment (Meiyim) [reply python/paddle/fluid/tests/unittests/npu/test_transpose_op_npu.py:55]: added

File: python/paddle/fluid/tests/unittests/npu/test_transpose_op_npu.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+
+
+@unittest.skipIf(not paddle.is_compiled_with_npu(),
+                 "core is not compiled with NPU")
+class TestTransposeOpFP16(OpTest):
+    no_need_check_grad = True 
+    def init_dtype(self):
+        self.dtype = np.float16
+
```
Comment (liym27) [inline python/paddle/fluid/tests/unittests/npu/test_transpose_op_npu.py:71]: TestTransposeOpFP16 中没有测试，可以参考 pow 单测写。另外需要提供调用 fp16 npu kernel 的日志截图。

Comment (Meiyim) [reply python/paddle/fluid/tests/unittests/npu/test_transpose_op_npu.py:71]: infoflow 2021-03-17 20-09-41

**Review by liym27 — APPROVED** [review]
LGTM

**Pull request merged**